src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkRedundantCommaInspection.kt

```diff
@@ -4,49 +4,62 @@ import com.intellij.codeInspection.*
 import com.intellij.openapi.command.WriteCommandAction
 import com.intellij.openapi.project.Project
 import com.intellij.psi.PsiElement
-import com.intellij.psi.PsiWhiteSpace
 import com.intellij.psi.TokenType
 import com.intellij.psi.util.elementType
-import com.intellij.psi.util.parents
-import com.intellij.psi.util.siblings
 import com.jetbrains.python.PyTokenTypes
 import com.jetbrains.python.psi.PyArgumentList
 import com.jetbrains.snakecharm.SnakemakeBundle
 import com.jetbrains.snakecharm.lang.psi.SmkArgsSection
-import com.jetbrains.snakecharm.lang.psi.SmkRuleOrCheckpointArgsSection
 
-class SmkRedundantComaInspection : SnakemakeInspection() {
+class SmkRedundantCommaInspection : SnakemakeInspection() {
     override fun buildVisitor(
         holder: ProblemsHolder,
         isOnTheFly: Boolean,
         session: LocalInspectionToolSession
     ) = object : SnakemakeInspectionVisitor(holder, session) {
 
-        private fun problemRegiser(psiEl: PsiElement) = holder.registerProblem(
-                psiEl,
-                SnakemakeBundle.message("INSP.NAME.redundant.coma.title"),
-                ProblemHighlightType.LIKE_UNUSED_SYMBOL,
-                fix
+        private fun problemRegiser(psiEl: PsiElement?) = psiEl?.let {
+            holder.registerProblem(
+                    it,
+                    SnakemakeBundle.message("INSP.NAME.redundant.coma.title"),
+                    ProblemHighlightType.LIKE_UNUSED_SYMBOL,
+                    fix
         )
+        }
 
-        private fun checkPyArgumentList(
+        private fun checkCommaInPyArgumentList(
                 element: PyArgumentList
-        ) {
-            element.children.reversed().forEach {
+        ): PsiElement? {
+
+            element.node.getChildren(null).reversed().forEach {
                 if (it.elementType !== PyTokenTypes.END_OF_LINE_COMMENT
                         && it.elementType !== PyTokenTypes.COMMA
-                        && it.elementType !== PyTokenTypes.WHITESPACE) return
-                else if (it.elementType == PyTokenTypes.COMMA)
-                    problemRegiser(it)
+                        && it.elementType !== TokenType.WHITE_SPACE) {
+                    return null
+                }
+                if (it.elementType == PyTokenTypes.COMMA) {
+                    return it.psi
+                }
             }
+
+            return null
         }
 
         override fun visitPyArgumentList(node: PyArgumentList) {
             if ((node.lastChild?.elementType !== PyTokenTypes.COMMA
                             && node.lastChild?.elementType !== PyTokenTypes.END_OF_LINE_COMMENT)
                     || node.parent !is SmkArgsSection) return
-            else if (node.lastChild?.elementType == PyTokenTypes.COMMA) problemRegiser(node.lastChild)
-            else if (node.lastChild?.elementType == PyTokenTypes.END_OF_LINE_COMMENT) checkPyArgumentList(node)
+
+            var problemEl: PsiElement? = null
```

Comment:
Usually better to use readonly variables (e.g. `val`) where possible, instead of `var`. E.g. here much shorter and elegant will be: 
```
val problemEl = when (node.lastChild?.elementType) {
   opt1 -> node.lastChild
   opt2 -> ...
   else ->  null
}

```

Reply:
Done